### PR TITLE
Prepare 0.24 release with branding change to Hickory DNS

### PR DESCRIPTION
**Pull request by bluejekyll**
See #2051

### CODE REVIEW

**Comment by bluejekyll** [comment]
@djc, I think this is ready, final release of trust-dns :(

I've tried to @ mention all the new names everywhere that I can.

**Comment by djc** [comment]
I put some feedback on the high-level plan in #2501.

**Comment by bluejekyll** [comment]
This won't be merged until after the repo is moved. I also need to update all brand names.

File: Cargo.toml
Comment (djc) [inline Cargo.toml:23]: Is it feasible to get a `hickory-dns` domain instead of one without the dash? Same for the GitHub org... IMO the lack of consistency is unfortunate.

Comment (bluejekyll) [reply Cargo.toml:23]: I've already registered that. We can absolutely use the `-`, we can do that for Organization as well. I wasn't honestly sure which is better for folks to use. The other option is to remove the `-` from the project name to make it consistent?

**Comment by bluejekyll** [comment]
Ok, I think this is ready @djc.

Open questions to resolve are basically on the names, essentially all the `-` or no `-` separators. I think for all the crates we should keep the `-` in the crate name. Now for the rest, should we setting on using `-` in the Organization and the Project, or no `-`? I think you want consistency. What's in this change now is this:

`https://github.com/hickorydns/hickory-dns` and `hickorydns.org`. I like the idea of being consistent, So I think the options are:

- `https://github.com/hickory-dns/hickory-dns` and `hickory-dns.org` or
- `https://github.com/hickorydns/hickorydns` and `hickorydns.org`

Pros for with the `-`, it will be more consistent with crate names, cons, it's not as "pretty" and may be less discoverable... But for that, we can just use redirects, or something.

What do you think?

File: Cargo.toml
Comment (djc) [reply Cargo.toml:23]: Personally I like having the `-`, visually separating words is pretty nice.

**Comment by djc** [comment]
> `https://github.com/hickorydns/hickory-dns` and `hickorydns.org`. I like the idea of being consistent, So I think the options are:
> 
>     * `https://github.com/hickory-dns/hickory-dns` and `hickory-dns.org` or
>     * `https://github.com/hickorydns/hickorydns` and `hickorydns.org`
> 
> Pros for with the `-`, it will be more consistent with crate names, cons, it's not as "pretty" and may be less discoverable... But for that, we can just use redirects, or something.
> 
> What do you think?

I like the `-` -- I find it more readable to have the words visually separated. It works with capitalization (HickoryDNS) but all lowercase (as in domain names or crate names) it's not as nice. So I would just call it Hickory-DNS and hickory-dns.

File: CHANGELOG.md
Comment (djc) [inline CHANGELOG.md:1]: I'd just expand the `f.k.a.`, might not be a well-known acronym for everyone.

**Review by djc — APPROVED** [review]

**Comment by bluejekyll** [comment]
> I like the `-` -- I find it more readable to have the words visually separated. It works with capitalization (HickoryDNS) but all lowercase (as in domain names or crate names) it's not as nice. So I would just call it Hickory-DNS and hickory-dns.

Fair, I anticipated some open discussion on this, so registered both variants. I think your reasoning is good here.

> For the crate names, are you going with hickory-dns-* or hickory-*? The shorter names are nice but also feel slightly inconsistent if the project name does include the DNS suffix. Not sure if that's a big issue.

Crate names are currently as follows:
- `trust-dns-proto` -> `hickory-proto`
- `trust-dns-resolver` -> `hickory-resolver`
- `trust-dns-recursor` -> `hickory-recursor`
- `trust-dns-server` -> `hickory-server`
- `trust-dns-client` -> `hickory-client`
- `trust-dns` -> `hickory-dns`

Personally, I like the shorter names, again, I registered both because I initially thought we might do `hickory-dns-*` for everything.

File: Cargo.toml
Comment (bluejekyll) [reply Cargo.toml]: ok, I updated.

File: CHANGELOG.md
Comment (bluejekyll) [reply CHANGELOG.md]: replaced with "formerly"

**Comment by bluejekyll** [comment]
Ok, I can't think of anything missing right now, I think everything is taken care of. Once this is approved, I will first merge in the 0.23.1 branch, and release that, then move the repo. Then merge this branch, and do a release on the hickory crates. assuming I got everything right that will just work. That could be delayed if I need to update any repo integration configs, etc. After that, over the weekend I can do things like properly wiring up the domain names, etc.

**Comment by bluejekyll** [comment]
oh, it looks like you approved, @djc, let me know if you have any concerns with the plan I mentioned above. I'll start that later today.

**Pull request merged**
